### PR TITLE
JS: recognize colon in command-prefixes for js/shell-command-constructed-from-input

### PR DESCRIPTION
**Pull request by erik-krogh**
Windows shell scripts sometimes use `:`. 

Gets a TP/TN for CVE-2017-1000219

### CODE REVIEW

File: javascript/ql/src/semmle/javascript/security/dataflow/UnsafeShellCommandConstructionCustomizations.qll [hunk 1]
```diff
@@ -85,7 +85,7 @@ module UnsafeShellCommandConstruction {
       this = root.getALeaf() and
       root = isExecutedAsShellCommand(DataFlow::TypeBackTracker::end(), sys) and
       exists(string prev | prev = this.getPreviousLeaf().getStringValue() |
-        prev.regexpMatch(".* ('|\")?[0-9a-zA-Z/]*")
+        prev.regexpMatch(".* ('|\")?[0-9a-zA-Z/:]*")
```
Comment (esbena) [inline javascript/ql/src/semmle/javascript/security/dataflow/UnsafeShellCommandConstructionCustomizations.qll:88]: While we are at it: do you see any reason not include `-` and `_` in the changed character class?

Comment (erik-krogh) [reply javascript/ql/src/semmle/javascript/security/dataflow/UnsafeShellCommandConstructionCustomizations.qll:88]: I don't see any reason not to do that. 
I'll add them.

**Review by esbena — CHANGES_REQUESTED** [review]
LGTM. Please add a change note.

File: change-notes/1.26/analysis-javascript.md [hunk 2]
```diff
@@ -30,6 +30,7 @@
 | Incomplete URL substring sanitization (`js/incomplete-url-substring-sanitization`) | More results | This query now recognizes additional URLs when the substring check is an inclusion check. |
 | Ambiguous HTML id attribute (`js/duplicate-html-id`) | Results no longer shown | Precision tag reduced to "low". The query is no longer run by default. |
 | Unused loop iteration variable (`js/unused-loop-variable`) | Fewer results | This query no longer flags variables in a destructuring array assignment that are not the last variable in the destructed array. |
+| Unsafe shell command constructed from library input (`js/shell-command-constructed-from-input`) | More results | This query now recognizes more commands where colon, slash, and underscore are used. |
```
Comment (esbena) [inline change-notes/1.26/analysis-javascript.md:33]:
```suggestion
| Unsafe shell command constructed from library input (`js/shell-command-constructed-from-input`) | More results | This query now recognizes more commands where colon, dash, and underscore are used. |
```

**Review by esbena — CHANGES_REQUESTED** [review]

**Review by esbena — APPROVED** [review]

**Pull request merged**